<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1658</width>
    <height>781</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(255, 255, 255);</string>
   </property>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1920</width>
      <height>1080</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI</family>
      <pointsize>10</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(0, 0, 0);
font: 10pt &quot;Segoe UI&quot;;
background-color: rgb(190, 190, 190);</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab_1">
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(190, 190, 190);</string>
     </property>
     <attribute name="title">
      <string>Tạo khóa</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>0</y>
        <width>731</width>
        <height>181</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(0, 0, 0);
</string>
      </property>
      <property name="title">
       <string>Chọn số nguyên tố</string>
      </property>
      <widget class="QLineEdit" name="NhapP">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>40</y>
         <width>113</width>
         <height>28</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);
placeholder-text-color: rgb(153, 153, 153);</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="placeholderText">
        <string>Nhập P</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="NhapQ">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>90</y>
         <width>113</width>
         <height>28</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);
placeholder-text-color: rgb(153, 153, 153);</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="placeholderText">
        <string>Nhập Q</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_P">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>16</width>
         <height>20</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Segoe UI</family>
         <pointsize>10</pointsize>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(0, 0, 0);</string>
       </property>
       <property name="text">
        <string>P</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_Q">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>80</y>
         <width>21</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Segoe UI</family>
         <pointsize>10</pointsize>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(0, 0, 0);</string>
       </property>
       <property name="text">
        <string>Q</string>
       </property>
      </widget>
      <widget class="QLabel" name="thongBaoLoiP">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>40</y>
         <width>113</width>
         <height>28</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(209, 0, 0);</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="thongBaoLoiQ">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>90</y>
         <width>113</width>
         <height>28</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(209, 0, 0);</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>190</y>
        <width>731</width>
        <height>481</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(0, 0, 0);</string>
      </property>
      <property name="title">
       <string>Tính các giá trị</string>
      </property>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>100</y>
         <width>321</width>
         <height>191</height>
        </rect>
       </property>
       <property name="title">
        <string>GroupBox</string>
       </property>
       <widget class="QLabel" name="label_title_3">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>-1</y>
          <width>131</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>10</pointsize>
          <italic>false</italic>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);</string>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Khóa công khai&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_E">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>80</y>
          <width>16</width>
          <height>20</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>10</pointsize>
          <italic>false</italic>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);</string>
        </property>
        <property name="text">
         <string>E</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_N">
        <property name="geometry">
         <rect>
          <x>220</x>
          <y>80</y>
          <width>21</width>
          <height>20</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>10</pointsize>
          <italic>false</italic>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);</string>
        </property>
        <property name="text">
         <string>N</string>
        </property>
       </widget>
       <widget class="QLabel" name="showE">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>120</y>
          <width>113</width>
          <height>28</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);
background-color: rgb(255, 255, 255);</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="showN">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>120</y>
          <width>113</width>
          <height>28</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);
background-color: rgb(255, 255, 255);</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </widget>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>100</y>
         <width>321</width>
         <height>191</height>
        </rect>
       </property>
       <property name="title">
        <string>GroupBox</string>
       </property>
       <widget class="QLabel" name="label_title_4">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>101</width>
          <height>20</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>10</pointsize>
          <italic>false</italic>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);</string>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Khóa bí mật&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_N_2">
        <property name="geometry">
         <rect>
          <x>220</x>
          <y>80</y>
          <width>21</width>
          <height>20</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>10</pointsize>
          <italic>false</italic>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);</string>
        </property>
        <property name="text">
         <string>N</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_D">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>80</y>
          <width>21</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <family>Segoe UI</family>
          <pointsize>10</pointsize>
          <italic>false</italic>
          <bold>false</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);</string>
        </property>
        <property name="text">
         <string>D</string>
        </property>
       </widget>
       <widget class="QLabel" name="showD">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>120</y>
          <width>113</width>
          <height>28</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);
background-color: rgb(255, 255, 255);</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="showN_2">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>120</y>
          <width>113</width>
          <height>28</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">color: rgb(0, 0, 0);
background-color: rgb(255, 255, 255);</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </widget>
      <widget class="QPushButton" name="buttonTaoKhoa">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>310</y>
         <width>101</width>
         <height>51</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(64, 127, 191);</string>
       </property>
       <property name="text">
        <string>Tạo khóa</string>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonExit">
       <property name="geometry">
        <rect>
         <x>500</x>
         <y>310</y>
         <width>101</width>
         <height>51</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(64, 127, 191);</string>
       </property>
       <property name="text">
        <string>Thoát</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>Chữ ký</string>
     </attribute>
     <widget class="QPushButton" name="taiVB_2">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>530</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Tải văn bản </string>
      </property>
     </widget>
     <widget class="QPushButton" name="Ky">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>530</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Ký</string>
      </property>
     </widget>
     <widget class="QPushButton" name="luuChuKy">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>530</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Lưu chữ ký</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>700</width>
        <height>500</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>399</width>
        <height>0</height>
       </size>
      </property>
      <property name="whatsThis">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:700;&quot;&gt;Ký&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>PHÁT SINH CHỮ KÝ</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <widget class="QLabel" name="labelVBCanKy_3">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>410</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Segoe UI</family>
         <pointsize>11</pointsize>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font: 11pt &quot;Segoe UI&quot;;
font-weight: bold</string>
       </property>
       <property name="text">
        <string>Chữ ký</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="chuKy">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>385</y>
         <width>421</width>
         <height>80</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);
font: 12pt &quot;Times New Roman&quot;;</string>
       </property>
      </widget>
      <widget class="QLabel" name="labelVBCanKy">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>140</y>
         <width>131</width>
         <height>20</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Segoe UI</family>
         <pointsize>11</pointsize>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font: 11pt &quot;Segoe UI&quot;;
font-weight: bold</string>
       </property>
       <property name="text">
        <string>Văn bản cần ký</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="VBCanKy">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>54</y>
         <width>421</width>
         <height>221</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);
font: 12pt &quot;Times New Roman&quot;;</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="buttonExit_2">
      <property name="geometry">
       <rect>
        <x>710</x>
        <y>650</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Thoát</string>
      </property>
     </widget>
     <widget class="QPushButton" name="taiChuKy">
      <property name="geometry">
       <rect>
        <x>1040</x>
        <y>540</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Tải chữ ký</string>
      </property>
     </widget>
     <widget class="QPushButton" name="xacNhan">
      <property name="geometry">
       <rect>
        <x>1160</x>
        <y>540</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Xác nhận</string>
      </property>
     </widget>
     <widget class="QPushButton" name="taiVB_3">
      <property name="geometry">
       <rect>
        <x>900</x>
        <y>540</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Tải Văn Bản</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_8">
      <property name="geometry">
       <rect>
        <x>780</x>
        <y>10</y>
        <width>700</width>
        <height>500</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>XÁC NHẬN CHỮ KÝ</string>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>140</y>
         <width>191</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">font: 700 11pt &quot;Segoe UI&quot;;</string>
       </property>
       <property name="text">
        <string>Văn bản cần xác nhận</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="VBCanXacNhan">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>54</y>
         <width>421</width>
         <height>221</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>12</pointsize>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);
font: 12pt &quot;Times New Roman&quot;;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>410</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">font: 700 11pt &quot;Segoe UI&quot;;</string>
       </property>
       <property name="text">
        <string>Xác nhận chữ ký</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="XacNhanChuKy">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>385</y>
         <width>421</width>
         <height>80</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Times New Roman</family>
         <pointsize>12</pointsize>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);
font: 12pt &quot;Times New Roman&quot;;</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="Chuyen">
      <property name="geometry">
       <rect>
        <x>580</x>
        <y>530</y>
        <width>90</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(64, 127, 191);</string>
      </property>
      <property name="text">
       <string>Chuyển</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>NhapP</tabstop>
  <tabstop>NhapQ</tabstop>
  <tabstop>buttonTaoKhoa</tabstop>
  <tabstop>buttonExit</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
